code/game/objects/items/melee/baton.dm

```diff
@@ -549,6 +550,16 @@
 	. = ..()
 	if (!(. & EMP_PROTECT_SELF))
 		deductcharge(1000 / severity)
+	if (cell.charge >= cell_hit_cost)
+		for(var/loops = 1, loops < rand(6,12), loops++)
+			scramble_mode()
```

Comment:
This is not what you want I think, this will spam call the proc by doing it every loop in addition to the callback. 

You prob want to put this before the loop (553-554)